airflow/providers/amazon/aws/utils/__init__.py

```diff
@@ -18,16 +18,29 @@
 
 import logging
 import re
+import warnings
 from datetime import datetime
 from enum import Enum
 
+from airflow.exceptions import AirflowProviderDeprecationWarning
+from airflow.utils.helpers import prune_dict
 from airflow.version import version
 
 log = logging.getLogger(__name__)
 
 
 def trim_none_values(obj: dict):
-    return {key: val for key, val in obj.items() if val is not None}
+    from packaging.version import Version
+
+    from airflow.version import version
+
+    if Version(version) < Version("2.7"):
+        return {key: val for key, val in obj.items() if val is not None}
+    else:
+        warnings.warn(
+            "use airflow.utils.helpers.prune_dict() instead", AirflowProviderDeprecationWarning, stacklevel=2
+        )
```

Comment:
idk if this is good...
because during the whole transition period, when users will have airflow >= 2.7 but we'll still have to support older core versions, they'll see this warning without being about to do anything to fix it (and we won't be able to fix it either)

Reply:
But they wont see this warning right? You are testing the Airflow version right before

Reply:
What I mean is that users that are up to date will see this warning because their version is going to be more recent than 2.6.

I'd like them to see this warning if it's their code calling the function (unlikely, but I understood that we cannot just remove a function before warning before) ; but NOT if it's called in an operator they are using, because then they cannot do anything about it.

Reply:
> but NOT if it's called in an operator they are using, because then they cannot do anything about it.

If your PR gets merged, they can upgrade to the newest Amazon provider package.

Reply:
yes, BUT we cannot update the method usage in the provider package until 1 year in the future, because current and near future versions of the provider package need to be back-compatible with older versions of core airflow (before this bugfix).

Reply:
so, someone that is perfectly up to date on all versions will see this warning for a full year (until backcompat for core airflow 2.6 is dropped and we can migrate usage in the provider package)

Reply:
After discussion, we came to a conclusion that it's better to remove that warning for now, and only uncover it once we'll be able to remove it from our own code.